app/services/parse/google_parse_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Parse
+  class GoogleParseService
+    SELECTORS = {
+      adwords_top: 'div.pla-unit-container a.pla-unit-title-link',
+      adwords_body: 'div.data-text-ad div.v5yQqb',
+      non_adwords: 'div#search div.yuRUbf a[data-ved]',
+      href: 'href'
+    }.freeze
+
+    def initialize(html_code)
+      @html_code = html_code
+      raise IcRailsHueyToby::Errors::ParseServiceError unless html_code
```

Comment:
Do we need this error? As we check the nil `html_code`, can we return nil in this case?
IMO, this error should be raised when there is something wrong with the parser 🤔

Reply:
If we want to add a check (which is not compulsory), we could rely on Nokogiri to tell us if there is any error (see https://nokogiri.org/tutorials/ensuring_well_formed_markup.html#gentle-well-formedness)

Reply:
@olivierobert By relying on Nokogiri to validate the `html_code`, even with the valid one, it's raising issues such as `Tag svg invalid` if the html_code contains svg tag and so on; when passing `nil`, it will skip the validations and no errors will be raised. 💭 It's so confusing if we rely on well-formed validations to raise the error if any.

Reply:
Understood. So let's raise an exceptopn then but I recommend to place it before local variales assignments and add a line break for readability:

```ruby
def initialize(html_code)
  raise IcRailsHueyToby::Errors::ParseServiceError unless html_code

  @html_code = html_code
end
```